### PR TITLE
Failing early if missing flux binary during cluster creation

### LINKED ISSUE #4516: Flux binary existence is not checked prior to running `create cluster` with flux
**What were you trying to accomplish?**
`eksctl create cluster -f cluster.yaml`

**What happened?**

`[!]  exec: "flux": executable file not found in $PATH`

**How to reproduce it?**
See above
It's obvious that flux will fail if it's not in path.

**Versions**

```
$ eksctl info
eksctl version: 0.76.0
kubectl version: v1.21.2-13+d2965f0db10712
OS: linux
```

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes #4516 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by Skarlso** [comment]
Not sure how to cover this with a test, since we don't really have anything to mock the os call and local environment does have flux... Maybe I can mess with the environment end clear the `PATH` variable....

**Comment by cPu1** [comment]
>Maybe I can mess with the environment end clear the `PATH` variable....

That might also result in other unintended side effects.

Adding a test for this isn't straightforward if we're using `exec.LookPath`, so I have no objections to not having a test for this change since the validation is quite straightforward.

**Comment by Skarlso** [comment]
@cPu1 What do you think about this... I added a small test in the gitops integration test which looks something like this:

```go
		FWhen("flux binary is not on path", func() {
			It("should fail early", func() {
				cmd := params.EksctlEnableCmd.WithArgs(
					"flux",
					"--config-file", "-",
				).WithStdin(clusterutils.Reader(clusterConfig))
				cmd = cmd.WithCleanEnv()
				session := cmd.Run()
				Expect(session.ExitCode()).To(Equal(1))
				Expect(string(session.Out.Contents())).To(ContainSubstring("asdf"))
			})
		})
```

So I'm using `WithCleanEnv` that should do the trick I believe. :) WDYT?

File: pkg/ctl/create/cluster.go [hunk 1]
```diff
@@ -195,6 +197,17 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 	if err := eks.ValidateFeatureCompatibility(cfg, kubeNodeGroups); err != nil {
 		return err
 	}
+
+	// Check if flux binary exists early in the process, so it doesn't fail at the end when the cluster
+	// has already been created with a missing flux binary error which should have been caught earlier.
+	// Note: we aren't running PreFlight here, we just check for the binary.
+	if cfg.HasGitOpsFluxConfigured() {
+		if _, err := exec.LookPath("flux"); err != nil {
+			logger.Warning(err.Error())
```
Comment (cPu1) [inline pkg/ctl/create/cluster.go:206]: This message may not make sense on its own. It'd be more meaningful to users if it's added to the error below.

File: pkg/ctl/create/cluster.go [hunk 2]
```diff
@@ -195,6 +197,17 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
...
+			return errors.New("flux binary is required when gitops configuration is set")
```
Comment (cPu1) [inline pkg/ctl/create/cluster.go:207]: Based on the suggestion above:
```suggestion
			return fmt.Errorf("flux binary is required when gitops configuration is set: %w", err)
```

Comment (Skarlso) [reply pkg/ctl/create/cluster.go:207]: Yeah sounds good

**Review by cPu1 — APPROVED** [review]
Quick fix! I left a minor suggestion but otherwise LGTM. :tada:

**Pull request merged**